### PR TITLE
added makefile, github-actions and updated go version

### PR DESCRIPTION
**Pull request by bharat-rajani**
## What type of PR is this?
- [x] Refactor
- [x] Optimization

## Description
This PR aims to 
- update go version to 1.20
- introduce make file
- configure golangci lint
- introduce github actions/workflows

## Added/updated tests?
- [x] Yes

## Run verification and test
- [x] `make verify` is passing
- [x] `make test` is passing

### CODE REVIEW

File: handlers_go18_test.go [hunk 1]
```diff
@@ -1,9 +1,10 @@
+//go:build go1.8
```
Comment (coreydaley) [inline handlers_go18_test.go:1]: Remove this file

**Review by coreydaley — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# Test binary, built with `go test -c`
+*.test
+
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Dependency directories (remove the comment below to include it)
+# vendor/
+
```
[the remark this answers is not in the document]
Comment (bharat-rajani) [reply .gitignore:28]: ACK. Is it okay if I keep IDE metadata directories in gitignore?
```
# IDE Metadata
.idea
.vscode

# Output of the go test coverage tool
coverage.coverprofile
```

[the remark this answers is not in the document]
Comment (coreydaley) [reply .gitignore]: Those should really be in a users global gitignore for whatever ide they use.

File: compress.go [hunk 3]
```diff
@@ -44,13 +44,13 @@ type flusher interface {
 	Flush() error
 }
 
-func (w *compressResponseWriter) Flush() {
+func (cw *compressResponseWriter) Flush() {
 	// Flush compressed data if compressor supports it.
-	if f, ok := w.compressor.(flusher); ok {
-		f.Flush()
+	if f, ok := cw.compressor.(flusher); ok {
+		_ = f.Flush()
```
Comment (coreydaley) [inline compress.go:50]: We should check the error here

File: compress.go [hunk 4]
```diff
@@ -44,13 +44,13 @@ type flusher interface {
...
 	}
 	// Flush HTTP response.
-	if f, ok := w.w.(http.Flusher); ok {
+	if f, ok := cw.w.(http.Flusher); ok {
 		f.Flush()
```
Comment (coreydaley) [inline compress.go:54]: We should check the error here

File: compress_test.go [hunk 5]
```diff
@@ -27,10 +26,10 @@ func compressedRequest(w *httptest.ResponseRecorder, compression string) {
 		w.Header().Set("Content-Length", strconv.Itoa(9*1024))
 		w.Header().Set("Content-Type", contentType)
 		for i := 0; i < 1024; i++ {
-			io.WriteString(w, "Gorilla!\n")
+			io.WriteString(w, "Gorilla!\n") //nolint:errcheck // this error is safe to ignore in unit test
```
Comment (coreydaley) [inline compress_test.go:29]: We should check the error

**Review by coreydaley — CHANGES_REQUESTED** [review]

File: compress.go (hunk 4 above)
Comment (bharat-rajani) [reply compress.go:54]: Do you mean we should handle the false codnition of `ok` (bool)?

Comment (bharat-rajani) [reply compress.go:54]: The http.Flusher.Flush won't return the error. Do you mean I should check the false case of `ok` (bool)?

File: compress.go (hunk 3 above)
Comment (bharat-rajani) [reply compress.go:50]: How should we handle the error? We can't return the error as `compressResponseWriter.Flush()` doesn't returns an error. I can use the inbuilt log package to log it, but I have some concerns. Would the default inbuilt logger print the error in a different format or manner than the user's configured logger?

File: compress.go (hunk 4 above)
Comment (coreydaley) [reply compress.go:54]: Apologies, I did not look close enough at this and thought it was similar to the above f.Flush().

File: compress.go (hunk 3 above)
Comment (coreydaley) [reply compress.go:50]: Yeah, I guess there is no other error logging in this package either
@apoorvajagtap do you have any thoughts on this?

File: README.md [hunk 6]
```diff
@@ -1,7 +1,8 @@
 gorilla/handlers
 ================
```
Comment (coreydaley) [inline README.md:2]:
```suggestion
# gorilla/handlers

```

**Review by coreydaley — CHANGES_REQUESTED** [review]

**Review by coreydaley — APPROVED** [review]

**Pull request merged**